text/0148-sampling-probability.md

```diff
@@ -0,0 +1,796 @@
+and before [Builtin
+Samplers](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#built-in-samplers),
+a new section will introduce the term "adjusted count" and relate it
+to inclusion probability.  For example:
+
+```markdown
+### Probability sampling
+
+Probability Samplers are Samplers that output statistically unbiased
+inclusion probability.  Inclusion probability in the context of
+tracing is the *effective* probability of the Sampler returning
+`RECORD_AND_SAMPLE` when invoked, which can be decided locally or 
+derived from the context when the [W3C Trace Context is-sampled
+flag](https://www.w3.org/TR/trace-context/#sampled-flag) is in use.
+
+#### Adjusted count span attribute
+
+The recommended way to convey inclusion probability *for events*
+sampled in OpenTelemetry is in the form of an **adjusted count**,
+which is the reciprocal (i.e., mathematical inverse function) of
+inclusion probability.
+
+The implied goal of probability sampling is to support estimating the
+count of spans in the population using the spans that were sampled.
+The adjusted count associated with a Span is the expected value of the
+number of identical Spans within the population that each Span
+represents.  Probability samplers SHOULD ensure the samples they
+compute are unbiased, which implies that the expected value of the sum
+of adjusted counts in the sample equals the true count of spans in the
+population.
+
+The adjusted count in an unbiased probability sampling scheme SHOULD
+be expressed using a Span attribute named `sampling.adjusted_count`
+when it represents the whole population of events.  Adjusted count 
+attributes MAY be integer or floating-point values.
+
+#### Inclusion probability tracestate value
+
+The recommended way to convey sampling probability *for contexts* in
+OpenTelemetry is through the W3C Trace Context tracestate using the
+key `head_probability`.
+
+Probability Samplers SHOULD encode the effective sampling inclusion
+probability using tracestate, for the context that was in effect when
+the W3C is-sampled bit was set.  The tracestate field SHOULD be set in
+both sampled and unsampled cases, to convey the inclusion probability
+even for unsampled contexts.
+
+The `head_probability` tracestate key is set to a floating-point
+number greater than or equal to 0 and less than or equal to 1.  The
+floating point precision of the number SHOULD follow implementation
+language standards and SHOULD be high enough to identify when Samplers
+have different inclusion probabilities.
+
+#### Counting probabilistically sampled spans
+
+Consumers of a stream of span data that may or may not have been
+sampled can follow these steps to count or approximately count the
+total number of spans in the population.
+
+For each span processed, locate the `sampling.adjusted_count` attribute. 
+If there is none, count a single span event.  If the attribute is set, 
```

Comment:
> If there is none, count a single span event.

I don't think this is correct, because spans inherit adjusted count from their parent

Reply:
There are several ways to interpret a count. We can count spans by the adjusted count of all the trace roots, or we can count spans by the adjusted count of all the spans independently. I aimed to talk about counting spans in their own right in this section, without referring to trace roots.

Spans may be counted using the adjusted count of the parent, in which case once in a while the parent's adjusted count comes through along with `is-sampled`. This also implies that much of the time `is-sampled` is not set and a zero adjusted count is assumed.

Spans may be counted by sampling at the call site independently, in which case if there is no sampling being applied, there will be no adjusted count set and the correct count is 1.

You are correct, I have failed to call out an error scenario, when `is-sampled` is set and the `otel=headprob:` value is unknown. I've written that the Parent Sampler MUST set `adjusted_count=0`, to signal that they must be counted using trace roots, not using span adjusted counts.

That way, when there is no sampling taking place, 100% of spans will be recorded and counted as one span each.